### ISSUE TITLE
Write Users cannot acknowledge or clear alarms [Bug] 

### ISSUE THREAD

**Issue by ianog-eng** [thingsboard/thingsboard#6370]
**Describe the bug**
Users who have write permission to a device group cannot acknowledge or clear alarms for those devices.

**Your Server Environment**
* cloud.thingsboard.io

**Your Client Environment**
**Desktop (please complete the following information):**

* OS: Windows 11 Home 22000.556
* Browser: Chrome
* Version: Version 99.0.4844.74 (Official Build) (64-bit)

**Your Device**

* Connectivity
  * Integration: AWS IoT Core 
* Custom device integrating RaspberryPi Zero W running AWS GreenGrass

**To Reproduce**
1. Create device
2. Add device to device group
3. Create user
4. Add user to user group
5. assign user group 'write user' role for device group
6. Generate device alarm
7. Use Alarm Widget to clear or acknowledge alarm
8. Get permission error

**Expected behavior**
The Alarm should be cleared or acknowledged.

**Screenshots**
*User Group Permissions*
![image](https://user-images.githubusercontent.com/84106368/161730495-f51083bc-3f1d-44a7-86a0-2a185cfd0954.png)

*Device Group Permissions*
![image](https://user-images.githubusercontent.com/84106368/161730713-9eed614a-e7c9-4a79-9a48-aa4a64d9f688.png)

*Error Message*
![image](https://user-images.githubusercontent.com/84106368/161730864-036f3db2-c33e-4a6c-ba56-6705cb46e1e5.png)

**Additional context**
May be related to recent PR affecting alarms permission checker... #6204

**Comment by ashaforost** [comment]
Hi @ianog-eng,


We did in fact have a bug related to the customers' alarms in the v3.3.4, however, we promptly released a bugfix for our cloud platform, as well as standalone. 
Also, please make sure that your customer has not only 'write' device permission, but read/write generic permission for the Alarm resource as well. 

Please let me know whether you still have this issue on v3.3.4.1

**Comment by daanstevenson** [comment]
Still seeing this issue as of today.
![image](https://user-images.githubusercontent.com/26634522/175424309-df28d302-896b-48f7-aedc-60031ddf542d.png)
Customer user group has 'Entity Group Write User' Role for the Device Group that the device is in, and 'Generic Alarm Resource All' Role (as well as for Assets and Devices, none seem to help).
![image](https://user-images.githubusercontent.com/26634522/175424615-35bd8754-80a2-4dd0-8f5f-be9194812a85.png)
If Device Ownership is transferred to Customer, alarms generated after transfer can be cleared and acknowledged by Customer.
Thanks!

**Comment by daanstevenson** [comment]
Still seeing this issue as of today.

![image](https://user-images.githubusercontent.com/26634522/175425205-a5294056-74c5-40dc-b900-f8898d03a09d.png)

Customer user group has 'Entity Group Write User' Role for the Device Group that the device is in, and 'Generic Alarm Resource All' Role (as well as for Assets and Devices, none seem to help).

![image](https://user-images.githubusercontent.com/26634522/175425054-4f0a682b-9ae7-418a-9333-1befde414ecc.png)

If Device Ownership is transferred to Customer, alarms generated after transfer can be cleared and acknowledged by Customer.
Thanks!

**Comment by nowherex** [comment]
We're having the same problem when trying to clear or acknowledge alarm of an customer "child " with a "parent" client.

We have a client that has other clients below, when an alarm is generated and propagated, we cannot do either of these two actions, only with the login of the tenant or the child client who owns the device, we know that it is not an access rule because we can do absolutely everything with the device/customer ( ALL function).

We tried everything, change the access role to generic/group, sharing the device with the parent client, etc., but without success.

**Comment by Civilduino** [comment]
I am having the same issue.
I as a Tenant own the devices and share them with my Costumers (e.g. device leasing). I share the Device Group using reading and writing roles. However, my costumer cannot acknowledge or clear the alarms, only I can do it. 

It seems that the Device ownership is a problem, could ownership be shared somehow, representing a leasing case?

**Comment by claezon** [comment]
We also need the ability for a Customer Administrator to be able to clear/acknowledge sub-customer alarms. Any updates regarding this?

**Comment by ashaforost** [comment]
@claezon @nowherex @netbc @ivojanalik @Civilduino @ianog-eng @daanstevenson ( I hope i included everyone here:))

Thingsboard v3.4.4 includes a redesign for this feature, so customers can now modify the alarms of their sub-customers. 

Please note that if you are a thingsboard cloud user, then this fix will be available in v3.5, as thingsboard cloud won't be upgraded to v3.4.4 just yet.

**Comment by GrayCygnus** [comment]
I recently needed to do this on v3.5.1 and at first didn't succeed. Eventually I stumbled with this thread and others and it seemed to be some loose ends still. I'm writing here what I did to achieve it and some observations.

> claezon nowherex netbc ivojanalik Civilduino ianog-eng daanstevenson ( I hope i included everyone here:))
> 
> Thingsboard v3.4.4 includes a redesign for this feature, so customers can now modify the alarms of their sub-customers.

This felt a bit like a loose end, cause this thread originally didn't ask for that (customers modifying the sub-customer alarms), as it asked instead for:

> Users who have write permission to a device group cannot acknowledge or clear alarms for those devices.

Which by the steps to reproduce and that description they are more about a "sub-customer" ack/clearing alarms from devices on a "upper level" (as in the tenant who has that customer, or sub-customer to customer, which is the other way around). 

So, after testing I managed to get this by doing an extra step that was missing from the steps OP listed: add to the User Group a "Generic Role" permission to the "Alarms" resource with read/write. 

![Screenshot from 2024-05-14 16-58-45](https://github.com/thingsboard/thingsboard/assets/7274943/f5ed91dd-7d1c-4e3d-a2fd-ca2050ff1a63)

In the end we have a [User]-[UserGroup]-[DeviceGroup]-[Device] relation, and the "bond" between the DeviceGroup and the UserGroup are the Roles we give the latter (a GroupRole to Read/Write the desired group and a GenericRole to Read/Write the resource Alarm).
